app.py

```diff
@@ -45,10 +45,13 @@ def inner_decorator(func):
         @wraps(func)
         def wrapped_func(*args, **kwargs):
             token = None
-            if request.method == 'GET':
+            token = request.cookies.get('token')
+            if token is None:
                 token = request.args.get('token')
-            elif request.method == 'POST':
-                token = request.values.get('token')
+                if token is not None:
+                    resp=redirect(f'{url_quote(request.base_url)}')
```

Comment:
Using f-strings you can format a variable inside a string. But here you are just putting one variable and nothing else inside the string, so the f-string part is unnecessary. `f'{X}' == X`

I don't think we want/need to url-escape the url we are redirecting to. So change the line to just `resp = redirect(request.base_url)`